cmp/grid/impl/ColumnGroupHeader.js

```diff
@@ -0,0 +1,72 @@
+import {div, span} from '@xh/hoist/cmp/layout';
+import {hoistCmp, HoistModel, useLocalModel} from '@xh/hoist/core';
+import {bindable} from '@xh/hoist/mobx';
+import {Icon} from '@xh/hoist/icon';
+import {isFunction} from 'lodash';
+import classNames from 'classnames';
+
+/**
+ * A custom ag-Grid group header component.
+ *
+ * @private
+ */
+export const ColumnGroupHeader = hoistCmp({
+    displayName: 'ColumnGroupHeader',
+    className: 'xh-grid-group-header',
+    model: false,
+
+    render(props) {
+        const impl = useLocalModel(() => new LocalModel(props)),
+            expandIcon = () => {
+                if (!impl.isExpandable) return null;
+
+                const icon = impl.isExpanded ? Icon.chevronLeft({size: 'sm'}) : Icon.chevronRight({size: 'sm'});
+                return div(icon);
+            };
+
+        // We will only have an xhColumnGroup if this was a configured column group. ag-Grid will
+        // auto-create column groups to ensure the headers are balanced for columns which are not
+        // in a group being rendered next to columns which are in groups.
+        let headerName = props.displayName;
+        const {xhColumnGroup, gridModel} = props;
+        if (xhColumnGroup && isFunction(xhColumnGroup.headerName)) {
+            headerName = xhColumnGroup.headerName({columnGroup: xhColumnGroup, gridModel});
+        }
+
+        return div({
+            className: classNames(props.className, impl.isExpandable ? 'xh-grid-group-header--expandable' : null),
+            items: [
+                span(headerName),
+                expandIcon()
+            ],
+            onClick: (e) => {
+                if (!impl.isExpandable) return;
+
+                e.stopPropagation();
+                props.setExpanded(!impl.isExpanded);
+            }
+        });
+    }
+});
+
+@HoistModel
+class LocalModel {
+    /** @member {ColumnGroup} */
```

Comment:
would not use this jsdoc for private class

Reply:
I've gotten into the habit of adding the `@member` doc for any member variables where the IDE cannot infer the type, I don't think of it so much as a doc but more of a type hint to make the dev experience better, especially when using types from third-party libs like in this case where it is an ag-Grid ColumnGroup. That being said I don't really care too much about it and happy to remove if we are trying to avoid any jsdoc-style  comments in private classes..

Reply:
not a big deal for this PR --- we can discuss our evolving standards on this issue later.

Reply:
Yes I occasionally add those @member declarations for the exact same reason, and have found them productive / useful to catch some errors.

Hopefully if we are able to introduce typescript into the equation we will get "real" help with these complex types.